src/main/services/token/TransferTokenService.kt

```diff
@@ -14,25 +14,39 @@ object TransferTokenService {
         caller: UserAccount,
         from: String,
         to: String,
-        name: String,
         amount: Double,
-        previousTransactionId: Int?,
-        notes: String?) : SOAResult<Transaction> {
+        name: String? = null,
+        tokenId: Int? = null,
+        previousTransactionId: Int? = null,
+        notes: String? = null) : SOAResult<Transaction> {
         val address = caller.cryptoKeyPair.publicKey
 
         // verify that the caller is the from address
         if(address != from)
             return SOAResult(SOAResultType.FAILURE, "Access denied. Caller and from address must match.", null)
 
-        return TransferTokenHelper.transferToken(
-            caller,
-            from,
-            to,
-            name,
-            amount,
-            ActionType.TRANSFER,
-            previousTransactionId,
-            notes
-        )
+        if (tokenId != null)
+            return TransferTokenHelper.transferToken(
```

Comment:
maybe we should refactor transferToken to handle this -- basically will have both optional params -- and error if both are null -- and will have a private method that handels getting the token